src/codegen/general.rs

```diff
@@ -909,6 +923,29 @@ pub fn doc_alias(w: &mut dyn Write, name: &str, comment_prefix: &str, indent: us
     )
 }
 
+pub fn external_doc_link(
+    w: &mut dyn Write,
+    external_url: Option<&str>,
+    name: &str,
+    visibility: &Visibility,
+    indent: usize,
+) -> Result<()> {
+    // Don't generate the external doc link on non-public types.
+    if !visibility.is_public() {
+        Ok(())
+    } else if let Some(external_url) = external_url {
+        writeln!(
+            w,
+            "{}/// This doc is limited due to license limitations. Please look at [our official docs]({}/index.html?search={}) for more information.",
```

Comment:
It's not because of a license issue, it's because docs.rs prevents us from generating documentation because the files are read-only.

Reply:
I guess it depends on how you see it. The reason the docs aren't in the generated code in the first place are license issues AFAIK, but of course that could be resolved if docs.rs would allow us to modify files.

So I suggest to write this instead: "This documentation is incomplete due to restrictions on docs.rs. Please look ..."

Reply:
Well, it's a license issue. That docs.rs doesn't support our strange use case is a secondary problem caused by the former.

But in any case, linking to the proper docs would be useful.

Reply:
Maybe we write: "This documentation is incomplete due to license restrictions and limitations on docs.rs. Please have a look at [our official docs]({}/index.html?search={})."